### PR TITLE
session permanence in db

### PR DESCRIPTION
**Pull request by HAQish**
need to think of more edge cases for testing, but it seems to exhibit basic functionality

### CODE REVIEW

File: frontend/src/App.js [hunk 1]
```diff
@@ -16,24 +16,22 @@ function  initializeReactGA(vals) {
...
     }
+    
     render() {
         return <div className="App">
             <Switch>
                 <Route path="/dashboard" component={DashboardContainer}/>
                 <AuthRoute path="/login" component={SessionFormContainer}/>
                 <Route exact path="/" component={Landing}/>
                 <Redirect to="/"/>
+                {/* <Redirect to="/placeholderURL2" /> */}
```
Comment (nucat176) [inline frontend/src/App.js]: why are we adding a commented out line here?

File: frontend/src/components/dashboard/challengeDetail.jsx [hunk 2]
```diff
@@ -28,19 +28,26 @@ export default class ChallengeDetail extends React.Component {
     }
 
     componentWillMount() {
+
+        // this.props.history.push("/login");
```
Comment (nucat176) [inline frontend/src/components/dashboard/challengeDetail.jsx]: same with here?

File: frontend/src/components/dashboard/myJobCents.jsx [hunk 3]
```diff
@@ -50,6 +53,15 @@ export default class MyJobCents extends React.Component {
         }
     }
 
+    imgLoadError(e) {
+        this.setState({ imageLoadErrBool: false });
+        e.target.src = ncentLogo;
+    }
+
+    imgLoad(e) {
+        // console.log("myJobCents.jsx image loaded successfully", e.target);
```
Comment (nucat176) [inline frontend/src/components/dashboard/myJobCents.jsx]: and here

File: frontend/src/components/dashboard/myJobCents.jsx [hunk 4]
```diff
@@ -89,13 +101,15 @@ export default class MyJobCents extends React.Component {
...
+                        onError={this.imgLoadError} onLoad={this.imgLoad}
+                    />
                     <h2 className="balance-title">{sponsoredChallenge.name}</h2>
                     <h3 className="balance-subtitle">Sponsored</h3>
-                    <h3 className="balance-subtitle">{sponsoredChallengeBalances[index]} jobCent(s)</h3>
+                    <h3 className="balance-subtitle">{ sponsoredChallengeBalances[index]  === 1 ? sponsoredChallengeBalances[index] + " jobCent" : sponsoredChallengeBalances[index] + " jobCents"}</h3>
                 </div>
                 <a
-                    title="New"
+                    title="Send jobCents to another user"
```
Comment (nucat176) [inline frontend/src/components/dashboard/myJobCents.jsx]: is this too verbose?

File: frontend/src/components/dashboard/myJobCents.jsx [hunk 5]
```diff
@@ -106,7 +120,7 @@ export default class MyJobCents extends React.Component {
                     Send
                 </a>
                 <a
-                    title="Redeem"
+                    title="Redeem a challenge"
```
Comment (nucat176) [inline frontend/src/components/dashboard/myJobCents.jsx]: also too verbose? how does it look on the tile?

File: frontend/src/components/dashboard/referralCode.jsx [hunk 6]
```diff
@@ -8,8 +8,12 @@ export default class ReferralCode extends React.Component {
         super(props);
     }
 
+    componentWillUnmount() {
+        window.history.replaceState({}, "jobCent", "/dashboard");
+    }
+
     render() {
-        console.log(this.props.referralCode);
+        // console.log(this.props.referralCode);
```
Comment (nucat176) [inline frontend/src/components/dashboard/referralCode.jsx]: commented

File: frontend/src/util/route_util.jsx [hunk 7]
```diff
@@ -9,6 +9,7 @@ const Auth = ({component: Component, path, loggedIn}) => (
         <Component {...props} />
       ) : (
         <Redirect to="/dashboard" />
+        // <Redirect to="/placeholderURL1" />
```
Comment (nucat176) [inline frontend/src/util/route_util.jsx]: and here

File: server/controllers/challengeUsers.js [hunk 8]
```diff
@@ -1,40 +1,56 @@
...
+                    challengeUuid: req.params.challengeUuid
+                }
+            });
+            if (!challengeUser) {
+                return res.status(404).send({message: "Referral Code not found"});
             }
-        });
-        if (!challengeUser) {
-            return res.status(404).send({message: "Referral Code not found"});
+            //Ned to Set the value wher
```
Comment (nucat176) [inline server/controllers/challengeUsers.js]: remove

File: frontend/src/components/dashboard/myJobCents.jsx (hunk 4 above)
Comment (HAQish) [reply frontend/src/components/dashboard/myJobCents.jsx]: I thought it was a bit more user-friendly.  The old title/tooltip was just "New" and it didn't really explain what the tab was doing.

File: frontend/src/components/dashboard/myJobCents.jsx (hunk 5 above)
Comment (HAQish) [reply frontend/src/components/dashboard/myJobCents.jsx]: I thought it was more explanatory.  If they hover their mouse there, they probably want details on what the button will do.

**Pull request merged**